handler/authorize.go

```diff
@@ -0,0 +1,71 @@
+package handler
+
+import (
+	"encoding/json"
+	"net/http"
+
+	"github.com/gitpods/gitpod"
+	"github.com/gitpods/gitpod/store"
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/go-kit/kit/metrics"
+	"golang.org/x/crypto/bcrypt"
+)
+
+const (
+	loginAttemptFailed  = "failed"
+	loginAttemptSuccess = "success"
+)
+
+var (
+	BadCredentialsJson = map[string]string{"message": "Bad credentials"}
+)
+
+type LoginStore interface {
+	GetUserByEmail(string) (gitpod.User, error)
+}
+
+type loginForm struct {
+	Email    string `json:"email"`
+	Password string `json:"password"`
+}
+
+func Authorize(logger log.Logger, loginAttempts metrics.Counter, s LoginStore) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		var form loginForm
+
+		defer r.Body.Close()
```

Comment:
Did know one can omit that call on the server. That happens when you use frameworks. 😁 
Thanks!